paper/src/main/java/com/denizenscript/denizen/paper/events/PlayerNameEntityScriptEvent.java

```diff
@@ -0,0 +1,95 @@
+package com.denizenscript.denizen.paper.events;
+
+import com.denizenscript.denizen.events.BukkitScriptEvent;
+import com.denizenscript.denizen.objects.EntityTag;
+import com.denizenscript.denizen.paper.PaperModule;
+import com.denizenscript.denizen.utilities.PaperAPITools;
+import com.denizenscript.denizen.utilities.implementation.BukkitScriptEntryData;
+import com.denizenscript.denizencore.objects.ObjectTag;
+import com.denizenscript.denizencore.objects.core.ElementTag;
+import com.denizenscript.denizencore.scripts.ScriptEntryData;
+import io.papermc.paper.event.player.PlayerNameEntityEvent;
+import net.md_5.bungee.api.ChatColor;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.Listener;
+
+public class PlayerNameEntityScriptEvent extends BukkitScriptEvent implements Listener {
+
+    // <--[event]
+    // @Events
+    // player names <entity>
+    //
+    // @Location true
+    //
+    // @Plugin Paper
+    //
+    // @Group Paper
+    //
+    // @Cancellable true
+    //
+    // @Triggers when a player attempts to rename an entity with a name tag.
+    //
+    // @Context
+    // <context.entity> returns an EntityTag of the renamed entity.
+    // <context.old_name> returns the old name of the entity, if any.
+    // <context.name> returns the new name of the entity.
+    // <context.persistent> returns whether this will cause the entity to persist through server restarts.
+    //
+    // @Determine
+    // "NAME:<ElementTag>" to set a different name for the entity.
+    // "PERSISTENT:<ElementTag(Boolean)>" to set whether the entity should remain through server restarts. ("True" keeps the entity across server restarts, "false" gets rid of the entity when the server restarts.)
```

Comment:
The way this was edited does not directly address my question: Does `false` on this cause the entity to not remain persistent, or does this stop the event from modifying the persistence state?